### DISCUSSION TITLE
small confusion

### DISCUSSION THREAD

**Discussion by algaloo** [metrumresearchgroup/mrgsolve#1024] [Q&A]
I am a tad bit confused on how to simulate from a model, say, 2 cmt linear, that is parameterized in microconstants and not CL and V terms. Can anyone give me an example where if I've fitted a model elsewhere in microconstants, I  can use mrgsolve to simulate?

**Comment by FelicienLL** [comment]
Hi @algaloo, you can code your model with differential equations and directly input your micro-constants within the `$ODE`. Alternatively,  if you want to use the built-in models in algebraic form (`$PKMODEL`) with micro-constants, you can reparameterize your model in terms of `CL`, `V1`, `Q` and `V2`.
``` r
library(mrgsolve)
#> 
#> Attachement du package : 'mrgsolve'
#> L'objet suivant est masqué depuis 'package:stats':
#> 
#>     filter
code <- "$PARAM K10 = 0.05, K12 = .1, K21 = 0.4, V = 30.0
$CMT CENT PERIPH
$ODE
dxdt_CENT = - (K12 + K10) * CENT + K21 * PERIPH ;
dxdt_PERIPH = K12 * CENT - K21 * PERIPH ;
$TABLE
capture CONC = CENT / V ; "
model <- mcode("model", code)
#> Building model ...
#> done.
model %>% 
  ev(amt = 1000) %>% 
  mrgsim() %>% 
  plot()
```

![](https://i.imgur.com/PsfpMjL.png)

``` r

code2 <- "$PARAM K10 = 0.05, K12 = .1, K21 = 0.4, V = 30.0
$CMT CENT PERIPH
$MAIN
double V1 = V ; 
double CL = K10 * V1 ; 
double Q = K12 * V1 ; 
double V2 = Q / K21 ;
$PKMODEL ncmt = 2, depot = FALSE
$TABLE
capture CONC = CENT / V1 ;
"
model2 <- mcode("model2", code2)
#> Building model2 ... done.
model2 %>% 
  ev(amt = 1000) %>% 
  mrgsim() %>% 
  plot()
```

![](https://i.imgur.com/vG8KQWo.png)

<sup>Created on 2022-10-11 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

I hope I answered your question

Félicien

**Comment by twa1970** [comment]
Hi @algaloo 

Please find enclosed for you info

https://www.ncbi.nlm.nih.gov/pubmed/32055585

Br

Thomas
